client/grpc_client.go

```diff
@@ -217,87 +240,115 @@ func (cli *grpcClient) finishAsyncCall(req *types.Request, res *types.Response)
 	return reqres
 }
 
+func (cli *grpcClient) checkErrGetResult() *types.Result {
+	if cli.err != nil {
+		errorLog := cli.err.Error()
+		cli.StopForError(cli.err)
+		result := types.ErrInternalError
+		result.SetLog(errorLog)
+		return &result
+	}
+	return nil
+}
+
+func (cli *grpcClient) checkGetErr() error {
+	if cli.err != nil {
+		err := errors.New(cli.err.Error())
+		cli.StopForError(cli.err)
+		return err
+	}
+	return nil
+}
+
 //----------------------------------------
 
 func (cli *grpcClient) EchoSync(msg string) (res types.Result) {
-	r := cli.EchoAsync(msg).Response.GetEcho()
-	return types.NewResultOK([]byte(r.Message), LOG)
+	reqres := cli.EchoAsync(msg)
```

Comment:
you mean because EchoAsync is actually a straight up synchronous call? it's kind of confusing either way